test/extensions/filters/http/json_to_metadata/filter_test.cc

```diff
@@ -1317,6 +1511,86 @@ TEST_F(FilterTest, RequestAllowEmptyContentType) {
   EXPECT_EQ(getCounterValue("json_to_metadata.rq_invalid_json_body"), 0);
 }
 
+TEST_F(FilterTest, ResponseAllowEmptyContentType) {
+  initializeFilter(R"EOF(
+response_rules:
+  rules:
+  - selectors:
+    - key: version
+    on_present:
+      metadata_namespace: envoy.lb
+      key: version
+  allow_empty_content_type: true
+)EOF");
+  const std::string response_body = R"delimiter({"version":"good version"})delimiter";
+  const std::map<std::string, std::string> expected = {{"version", "good version"}};
+
+  Http::TestResponseHeaderMapImpl matched_incoming_headers{{":path", "/ping"}, {":method", "GET"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_->encodeHeaders(matched_incoming_headers, false));
+
+  EXPECT_CALL(encoder_callbacks_, streamInfo()).WillRepeatedly(ReturnRef(stream_info_));
+  EXPECT_CALL(stream_info_, setDynamicMetadata("envoy.lb", MapEq(expected)));
+  testResponseWithBody(response_body);
+
+  EXPECT_EQ(getCounterValue("json_to_metadata.resp_success"), 1);
+  EXPECT_EQ(getCounterValue("json_to_metadata.resp_mismatched_content_type"), 0);
+  EXPECT_EQ(getCounterValue("json_to_metadata.resp_no_body"), 0);
+  EXPECT_EQ(getCounterValue("json_to_metadata.resp_invalid_json_body"), 0);
+}
+
+TEST_F(FilterTest, RequestBodyWithResponseRule) {
+  initializeFilter(R"EOF(
+response_rules:
+  rules:
+  - selectors:
+    - key: version
+    on_present:
+      metadata_namespace: envoy.lb
+      key: version
+      type: STRING
+)EOF");
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(incoming_headers_, false));
+
+  const std::string request_body = R"delimiter({"version":"good version"})delimiter";
+
+  testRequestWithBody(request_body);
```

Comment:
this should be end_stream false since we got a trailer. same in `ResponseBodyWithRequestRule`
Could we have a new test with multiple chuck data without trailer

Reply:
> this should be end_stream false since we got a trailer. same in `ResponseBodyWithRequestRule` Could we have a new test with multiple chuck data without trailer

there is EncodeTwoDataStreams should cover multiple chunk data without trailer